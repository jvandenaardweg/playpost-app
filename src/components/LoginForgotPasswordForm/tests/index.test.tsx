import React from 'react';
import { render, RenderAPI } from 'react-native-testing-library';

import { LoginForgotPasswordForm } from '../index';

describe('LoginForgotPasswordForm', () => {

  describe('rendering', () => {
    let wrapper: RenderAPI;

    beforeAll(() => {
      wrapper = render(<LoginForgotPasswordForm onChangeText={() => {}} onPressResetPassword={() => {}} onPressResetPasswordCode={() => {}} isSuccess={false} email="" isLoading={false} />);
    });

    it('should render correctly', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });
});
