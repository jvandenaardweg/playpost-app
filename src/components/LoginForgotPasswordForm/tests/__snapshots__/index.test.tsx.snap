// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForgotPasswordForm rendering should render correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#EEEEEE",
        "flex": 1,
        "justifyContent": "center",
        "padding": 12,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
  testID="LoginForgotPasswordForm"
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
    keyboardShouldPersistTaps="handled"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View>
      <TextInput
        autoCapitalize="none"
        autoFocus={false}
        blurOnSubmit={false}
        clearButtonMode="always"
        editable={true}
        keyboardType="email-address"
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="E-mail address"
        returnKeyType="next"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#e5e5e5",
            "borderRadius": 4,
            "borderWidth": 1,
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
            "height": 54,
            "lineHeight": 20,
            "marginBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
          }
        }
        testID="LoginForgotPasswordForm-TextInput-email"
        textContentType="username"
        value=""
      />
      <View>
        <View
          style={
            Object {
              "borderRadius": 3,
            }
          }
        >
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="LoginForgotPasswordForm-Button-reset-password"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#000000",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 12,
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": 18,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Reset my password
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "borderRadius": 3,
            }
          }
        >
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="LoginForgotPasswordForm-Button-reset-password-code"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#2089dc",
                    "fontSize": 18,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Already have reset password code?
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
